assets/js/modules/tagmanager/components/common/FormInstructions.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Tag Manager Form Instructions component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME as CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import { STORE_NAME } from '../../datastore/constants';
+const { useSelect } = Data;
+
+export default function FormInstructions() {
+	const hasExistingTag = useSelect( ( select ) => select( STORE_NAME ).hasExistingTag() );
+	const isSecondaryAMP = useSelect( ( select ) => select( CORE_SITE ).isSecondaryAMP() );
+
+	if ( hasExistingTag ) {
```

Comment:
It's only really relevant here since `ExistingTagNotice` returns null if `! hasExistingTag` which includes `undefined`.

I took a look at changing this and I think it's preferable to keep it like this because returning `null` early will cause a flicker every time due to the latency of the tag check request. This would cause the instructions to appear out of nowhere and shift the form down all of a sudden. So I think it's preferable for this flicker to only happen when the message changes due to the presence of an existing tag (which should be much less frequent than when one is present) rather than showing nothing at all until the tag fetch is resolved.

I considered adding our `PreviewBlock` to hold the place as well since some message will always appear, but it was not a great experience so I opted to keep it like this for now. Not saying this is the best way to do it, but I think it's the best way within the scope of this issue.